processing/src/main/java/io/druid/query/filter/BoundDimFilter.java

```diff
@@ -0,0 +1,172 @@
+import com.metamx.common.StringUtils;
+
+import java.nio.ByteBuffer;
+
+public class BoundDimFilter implements DimFilter
+{
+  private final String dimension;
+  private final String upper;
+  private final String lower;
+  private final boolean lowerStrict;
+  private final boolean upperStrict;
+  private final boolean alphaNumeric;
+
+  @JsonCreator
+  public BoundDimFilter(
+      @JsonProperty("dimension") String dimension,
+      @JsonProperty("lower") String lower,
+      @JsonProperty("upper") String upper,
+      @JsonProperty("lowerStrict") Boolean lowerStrict,
+      @JsonProperty("upperStrict") Boolean upperStrict,
+      @JsonProperty("alphaNumeric") Boolean alphaNumeric
+  )
+  {
+    this.dimension = Preconditions.checkNotNull(dimension, "dimension can not be null");
+    Preconditions.checkState((lower != null) || (upper != null), "lower and upper can not be null at the same time");
+    this.upper = upper;
+    this.lower = lower;
+    this.lowerStrict = (lowerStrict == null) ? false : lowerStrict;
+    this.upperStrict = (upperStrict == null) ? false : upperStrict;
+    this.alphaNumeric = (alphaNumeric == null) ? false : alphaNumeric;
+  }
+
+  @JsonProperty
+  public String getDimension()
+  {
+    return dimension;
+  }
+
+  @JsonProperty
+  public String getUpper()
+  {
+    return upper;
+  }
+
+  @JsonProperty
+  public String getLower()
+  {
+    return lower;
+  }
+
+  @JsonProperty
+  public boolean isLowerStrict()
+  {
+    return lowerStrict;
+  }
+
+  @JsonProperty
+  public boolean isUpperStrict()
+  {
+    return upperStrict;
+  }
+
+  @JsonProperty
+  public boolean isAlphaNumeric()
+  {
+    return alphaNumeric;
+  }
+
+  @Override
+  public byte[] getCacheKey()
+  {
+    byte[] dimensionBytes = StringUtils.toUtf8(this.getDimension());
+    byte[] lowerBytes = this.getLower() == null ? new byte[0] : StringUtils.toUtf8(this.getLower());
+    byte[] upperBytes = this.getUpper() == null ? new byte[0] : StringUtils.toUtf8(this.getUpper());
+    byte boundType = 0x1;
+    if (this.getLower() == null) {
+      boundType = 0x2;
+    } else if (this.getUpper() == null) {
+      boundType = 0x3;
+    }
+
+    byte lowerStrictByte = (this.isLowerStrict() == false) ? 0x0 : (byte) 1;
+    byte upperStrictByte = (this.isUpperStrict() == false) ? 0x0 : (byte) 1;
+    byte AlphaNumericByte = (this.isAlphaNumeric() == false) ? 0x0 : (byte) 1;
+
+    ByteBuffer boundCacheBuffer = ByteBuffer.allocate(
+        8
+        + dimensionBytes.length
+        + upperBytes.length
+        + lowerBytes.length
+    );
+    boundCacheBuffer.put(DimFilterCacheHelper.BOUND_CACHE_ID)
+                    .put(boundType)
+                    .put(upperStrictByte)
+                    .put(lowerStrictByte)
+                    .put(AlphaNumericByte)
+                    .put((byte) 0xFF)
+                    .put(dimensionBytes)
+                    .put((byte) 0xFF)
+                    .put(upperBytes)
+                    .put((byte) 0xFF)
```

Comment:
I think another PR introduced a separate variable for 0xFF in this context, can you use that?